packages/core/src/components/axes/axis.ts

```diff
@@ -70,13 +73,48 @@ export class Axis extends Component {
 				break;
 		}
 
+		// Add axis into the parent
+		const container = DOMUtils.appendOrSelect(svg, `g.axis.${axisPosition}`);
+		const axisRefExists = !container.select(`g.ticks`).empty();
+		let axisRef = DOMUtils.appendOrSelect(container, `g.ticks`);
+		if (!axisRefExists) {
+			axisRef.attr("role", `${Roles.GRAPHICS_OBJECT} ${Roles.GROUP}`);
+		}
+
+		// We draw the invisible axis because of the async nature of d3 transitions
+		// To be able to tell the final width & height of the axis when initiaing the transition
+		// The invisible axis is updated instantly and without a transition
+		const invisibleAxisRef = DOMUtils.appendOrSelect(container, `g.ticks.invisible`)
+			.style("opacity", "0")
+			.attr("aria-hidden", true);
+
+		// Append to DOM a fake tick to get the right computed font height
+		const fakeTick = DOMUtils.appendOrSelect(invisibleAxisRef, `g.tick`);
+		const fakeTickText = DOMUtils.appendOrSelect(fakeTick, `text`).text("0");
+		const tickHeight = DOMUtils.getSVGElementSize(fakeTickText.node(), { useBBox: true }).height;
+		fakeTick.remove();
+
 		// Initialize axis object
 		const axis = axisFunction(scale)
 			.tickSizeOuter(0)
 			.tickFormat(Tools.getProperty(axisOptions, "ticks", "formatter"));
 
 		if (scale.ticks) {
-			const numberOfTicks = Tools.getProperty(axisOptions, "ticks", "number") || Configuration.axis.ticks.number;
+			let numberOfTicks;
+
+			if (isNumberOfTicksProvided) {
+				numberOfTicks = numberOfTicksProvided;
+			} else {
```

Comment:
A bit of nitpicking:

It looks like this can be either that number of ticks is provided, or that it's a vertical axis, or that it's the default number of ticks. That would make this block look like

```javascript
if (isNumberOfTicksProvided) {
	// ...
} else if (isVerticalAxis) {
	// ...
} else {
	// ...
}
```

or that we set the default (since it's just a simple assignment) and then conditionally overwrite like

```javascript
numberOfTicks = Configuration.axis.ticks.number;
if (isNumberOfTicksProvided) {
	// ...
} else if (isVerticalAxis) {
	// ...
}
```

or to merge the ideas

```javascript
if (isNumberOfTicksProvided) {
	// ...
} else {
	numberOfTicks = Configuration.axis.ticks.number;
	if (isVerticalAxis) {
		// ...
	}
}
```

Reply:
Hi @zvonimirfras, 
I've refactored the code as the 3rd idea! The if statement should be more readable now. 
Thank you.